src/eiffel_graphql_api/graphql/schemas/links/artifact.py

```diff
@@ -43,9 +43,45 @@ def resolve_artifact_created(self, _):
         return ArtifactCreated(event)
 
 
-class ReusedArtifact(Artifact):
+class ReusedArtifact(graphene.ObjectType):
     """Reused artifact link."""
 
+    artifact_created = graphene.Field(ArtifactCreated)
+
+    def __init__(self, link):
+        """Initialize link."""
+        # pylint:disable=super-init-not-called
+        self.link = link
+
+    def resolve_artifact_created(self, _):
+        """Resolve artifact created link."""
+        from ..union import NotFound  # pylint:disable=import-outside-toplevel
+
+        event = find_one(
+            "EiffelArtifactCreatedEvent", {"meta.id": self.link.get("target")}
+        )
+        if event is None:
+            return NotFound(self.link, "Could not find event in database.")
+        return ArtifactCreated(event)
+
 
-class ArtifactPreviousVersion(Artifact):
+class ArtifactPreviousVersion(graphene.ObjectType):
     """Previous artifact version link."""
+
+    artifact_created = graphene.Field(ArtifactCreated)
+
+    def __init__(self, link):
+        """Initialize link."""
+        # pylint:disable=super-init-not-called
+        self.link = link
+
+    def resolve_artifact_created(self, _):
+        """Resolve artifact created link."""
+        from ..union import NotFound  # pylint:disable=import-outside-toplevel
+
+        event = find_one(
+            "EiffelArtifactCreatedEvent", {"meta.id": self.link.get("target")}
+        )
+        if event is None:
+            return NotFound(self.link, "Could not find event in database.")
+        return ArtifactCreated(event)
```

Comment:
Is this related to the package upgrade or something you did on the fly?

Reply:
It's related to the upgrade, for some reason having the ArtifactPreviousVersion and ReusedArtifact inherit from Artifact stopped working. I could perhaps have spent some time trying to figure out exactly why, but the pain of having a small boiler plate was way less discomforting than digging into the internals of the GraphQL libraries.

Reply:
No further luck, leaving it as it is for now.